### DISCUSSION TITLE
How to Specify Private Registries?

### DISCUSSION THREAD

**Discussion by GiovanniCampaner** [testcontainers/testcontainers-java#5053] [Q&A] [answered]
Coming from: https://github.com/testcontainers/testcontainers-java/issues/1689

I'm using Maven with JUnit 5, trying to setup a container like this:
```java
@QuarkusTest
@Testcontainers
@Tag("integration")
public class MyServiceIntegrationTest {

    private static final DockerImageName image = DockerImageName.parse("my-custom-repository/database/mariadb:latest")
            .asCompatibleSubstituteFor("mariadb");

    @Container
    private final MariaDBContainer<?> topLevelContainer = new MariaDBContainer<>(image)
            .withExposedPorts(3306);

    ...
}
```
But when launching with `mvn integration-test` I'm getting:
```
com.github.dockerjava.api.exception.InternalServerErrorException: 
Status 500: {"message":"Get https://registry-1.docker.io/v2/: Proxy Authentication Required"}
```
I can't use public repositories, how can I specify the image to be pulled from `my-custom-repository` and not `registry-1.docker.io` ?

**Comment by kiview** [comment, marked as answer]
This works already if you specify your own registry with its domain, see our docs:
https://www.testcontainers.org/features/image_name_substitution/#manual-substitution

**Reply by obfischer** [reply]
The mentioned documentation helped to solve the same issue for me. Therefore I think we can treat this question as answered.

**Comment by bademux** [comment]
For my isolated CICD i use `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX`

**Reply by FieteO** [reply]
How would authentication look like for this approach? Is there a `TESTCONTAINERS_HUB_USERNAME`,`TESTCONTAINERS_HUB_PASSWORD` variable that could be defined?

**Reply by FieteO** [reply]
Oh, I see this could be relevant for authentication: [Docker registry authentication](https://java.testcontainers.org/supported_docker_environment/#docker-registry-authentication)
